packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -453,8 +453,10 @@ export const RemoteServerFactory = {
     _setForwardedHeaders(app, options) {
         app.use((req, res, next) => {
             const xForwardedHost = req.headers?.['x-forwarded-host']
+            const xForwardedProto = req.headers?.['x-forwarded-proto']
             if (xForwardedHost) {
-                res.locals.xForwardedHost = xForwardedHost
+                // prettier-ignore
+                res.locals.xForwardedOrigin = `${xForwardedProto || options.protocol}://${xForwardedHost}`
```

Comment:
Is it supposed to be `req.protocol`? Based on a comment from last time.

```suggestion
                res.locals.xForwardedOrigin = `${xForwardedProto || req.protocol}://${xForwardedHost}`
```

Reply:
@vmarta the req.protocol seems to be `http` and it will break MRT as it will reject http and return 405. Please see the thread [here](https://salesforce-internal.slack.com/archives/C07R093Q3ME/p1728345854461289?thread_ts=1728076255.990489&cid=C07R093Q3ME)

Reply:
Consider a situation where
- `xForwardedHost` is undefined
- `options.protocol` is `http` (this is retail-react-app default)
- app is deployed on MRT

On MRT, will there be an issue that the generated origin is `http://...` unexpected since we want to use `https` on MRT?

Reply:
Can you please test the above scenario? 🙏

Reply:
@kevinxh if xForwardedHost is not defined, we are not passing it down to res.locals though. In this case, useOrigin will use APP_ORIGIN

Reply:
@kevinxh If you see this line [here](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/develop/packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js#L145). `options.procotol` is always https on remote server

Reply:
Ok, I finally see the full picture now... It's not that obvious in the code where the condition is implicit defined in two separate places.

I feel like it's more readable just remove `_setForwardedHeaders` and move the logic to `useOrigin`

Reply:
That would come back to the argument we had in slack where we have know when it is the devServer to use `http`. 

`option.protol` will set `http` for localhost and use `https` for remote server. 🤔

Reply:
right, you need access to `options`!